### PR TITLE
added aws-ci

### PR DESCRIPTION
**Pull request by germanilia**
new ci pipeline to publish docker images to aws ecr

### CODE REVIEW

File: .github/workflows/ci-aws.yml [hunk 1]
```diff
@@ -0,0 +1,259 @@
+name: CI-AWS
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [master]
```
Comment (mshidlov) [inline .github/workflows/ci-aws.yml:6]: should be all branches

File: .github/workflows/ci-aws.yml [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    paths-ignore:
+      - "**.md"
+      - ".github/workflows/**.yml"
+      - "renovate.json"
+      - ".all-contributorsrc"
+  pull_request:
+    branches: [master]
```
Comment (mshidlov) [inline .github/workflows/ci-aws.yml:13]: should be release/*

File: .github/workflows/ci-aws.yml [hunk 3]
```diff
@@ -0,0 +1,259 @@
+name: CI-AWS
```
Comment (mshidlov) [inline .github/workflows/ci-aws.yml:1]: Missing Discord notifications on failure

File: .github/workflows/ci-aws.yml [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }} 
+          aws-region: us-east-1
+          
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Set ecr repository
+        run: |
```
Comment (mshidlov) [inline .github/workflows/ci-aws.yml:170]: I need more information about this step

File: .github/workflows/ci-aws.yml [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+          sh .github/workflows/scripts/get_ecr_repository.sh
+        env:
+          SERVICE_NAME: ${{ matrix.service }}
+      - name: Set hash
+        env:
+          working-directory: packages/${{ matrix.service }}
+        run: |
+          GITHUB_HASH_MOD=$(find ./packages/${{ matrix.service }} -type f \( -exec sha1sum {} \; \) | awk '{print $1}' | sort | sha1sum)
```
Comment (mshidlov) [inline .github/workflows/ci-aws.yml:178]: github actions has a hash function hashFiles

File: .github/workflows/ci-aws.yml [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+          fi
+      
+      - name: set branch-name
+        run: |
+          if [[ $GITHUB_REF != 'refs/heads/master' ]]
+          then 
+            echo "BRANCH_NAME=$(echo $GITHUB_REF_NAME | sed "s/\//-/g")" >> $GITHUB_ENV
+          fi
+
+      - name: Build, tag, and push image to Amazon ECR
```
Comment (mshidlov) [inline .github/workflows/ci-aws.yml:207]: there is GH action that performs that step why not using it (it will be more readable and maintained)

File: .github/workflows/ci-aws.yml [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        run: |
+          Docker_image=$ECR_REGISTRY/$ECR_REPOSITORY:$GITHUB_HASH
+          docker build -f ./packages/${{ matrix.service }}/Dockerfile . -t $Docker_image
+          docker push $Docker_image
+          MANIFEST=$(aws ecr batch-get-image --repository-name $ECR_REPOSITORY --image-ids imageTag=$GITHUB_HASH --output json | jq --raw-output --join-output '.images[0].imageManifest')
+          aws ecr put-image --repository-name $ECR_REPOSITORY --image-tag ${{ github.sha }} --image-manifest "$MANIFEST"
+          aws ecr put-image --repository-name $ECR_REPOSITORY --image-tag ${{ github.ref_name }} --image-manifest "$MANIFEST"
+          aws ecr put-image --repository-name $ECR_REPOSITORY --image-tag ${{ github.run_id }} --image-manifest "$MANIFEST"          
+  docker-retag:
```
Comment (mshidlov) [inline .github/workflows/ci-aws.yml:219]: not clear why we have this and `Build, tag, and push image to Amazon ECR` step

File: packages/amplication-server/Dockerfile [hunk 8]
```diff
@@ -11,7 +11,7 @@ RUN npm config set fund false
 
 WORKDIR /app
 COPY lerna.json /app
-COPY package*.json /app
+COPY package*.json /app/
```
Comment (mshidlov) [inline packages/amplication-server/Dockerfile:14]: why did you change it?

File: packages/amplication-client/Dockerfile [hunk 9]
```diff
@@ -12,7 +12,7 @@ RUN npm config set fund false
 
 WORKDIR /app
 COPY lerna.json /app
-COPY package*.json /app
+COPY package*.json /app/
```
Comment (mshidlov) [inline packages/amplication-client/Dockerfile:15]: why did you change it?

**Comment by mshidlov** [comment]
the helm readme file is missing can you add it please?

File: .github/workflows/ci-aws.yml (hunk 1 above)
Comment (germanilia) [reply .github/workflows/ci-aws.yml:6]: fixed

File: .github/workflows/ci-aws.yml (hunk 2 above)
Comment (germanilia) [reply .github/workflows/ci-aws.yml:13]: fixed

File: .github/workflows/ci-aws.yml (hunk 7 above)
Comment (germanilia) [reply .github/workflows/ci-aws.yml:219]: fixed

File: .github/workflows/ci-aws.yml (hunk 6 above)
Comment (germanilia) [reply .github/workflows/ci-aws.yml:207]: I saw it, not sure how it works, if it's not very critical I would like to use our code where I can control exactly what happens.

File: .github/workflows/ci-aws.yml (hunk 3 above)
Comment (germanilia) [reply .github/workflows/ci-aws.yml:1]: done

File: .github/workflows/ci-aws.yml (hunk 4 above)
Comment (germanilia) [reply .github/workflows/ci-aws.yml:170]: fixed

File: .github/workflows/ci-aws.yml (hunk 5 above)
Comment (germanilia) [reply .github/workflows/ci-aws.yml:178]: as far as i understand hashFiles is for files, we need to hash the entire folder
